tests/rbd/test_rbd_immutable_cache.py

```diff
@@ -0,0 +1,225 @@
+        execute_command(client_node, f"chmod 777 {ceph_conf_path}")
+
+        for key, value in client_config.items():
+            command = f"ceph config set client {key} {value}"
+            execute_command(client_node, command)
+
+        log.info(
+            f"Updated the client configuration for immutable cache in {ceph_conf_path}"
+        )
+
+        # Install ceph-immutable-object-cache package
+        execute_command(client_node, "dnf install ceph-immutable-object-cache -y")
+
+        # Generate keyring and create keyring file
+        keyring_file = (
+            f"/etc/ceph/ceph.client.ceph-immutable-object-cache.{user_name}.keyring"
+        )
+        execute_command(client_node, f"touch {keyring_file}")
+        execute_command(client_node, f"chmod 777 {keyring_file}")
+
+        keyring_command = (
+            f"ceph auth get-or-create client.ceph-immutable-object-cache.{user_name}"
+            " mon 'profile rbd'"
+            " osd 'profile rbd-read-only'"
+        )
+        command = f"{keyring_command} >> {keyring_file}"
+        execute_command(client_node, command)
+
+        log.info(f"Keyring file {keyring_file} created and populated.")
+
+        # Enable and start the ceph-immutable-object-cache service
+        service_name = (
+            f"ceph-immutable-object-cache@ceph-immutable-object-cache.{user_name}"
+        )
+        execute_command(client_node, f"systemctl enable {service_name}")
+        # we need to wait for some time before starting daemon
+        time.sleep(60)
+        execute_command(client_node, f"systemctl start {service_name}")
+
+        # Check the status of the service
+        status_command = f"systemctl status {service_name}"
+        status_output = execute_command(client_node, status_command)
+
+        # Verify object cache daemon got activated successfully
+        if "Active" not in status_output:
+            log.debug(status_output)
+            log.error(f"{service_name} not activated")
+            return 1
+        log.info("ceph-immutable-object-cache daemon got activated successfully")
+
+    except Exception as err:
+        log.error(err)
+        return 1
+
+    return 0
+
+
+def verify_cache_objects(rbd, node, **kw):
+    """
+    Method to validate cache objects after configuring immutable cache daemon.
+
+    Args:
+        rbd: RBD object
+        node: cache client node
+        **kw: test data
+
+    Returns:
+        int: The return value. 0 for success, 1 for failure.
+    """
+    config = kw.get("config")
+    pool = config["rep_pool_config"]["pool"]
+    image = config["rep_pool_config"]["image1"]
+    image_size = config["rep_pool_config"]["size"]
+
+    snap_name = f"{pool}/{image}@snap1"
+    clone = "cloned_image1"
+    try:
+        rbd.create_image(
+            pool,
+            image,
+            size=image_size,
+        )
+
+        run_fio(image_name=image, pool_name=pool, client_node=node)
```

Comment:
Are you using the runtime passed from test suite to run fio?

Reply:
used the passed config variable.